application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,261 @@
+            event.reply("You dont have the right permissions to use this command")
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        Member selfMember = Objects.requireNonNull(event.getGuild()).getSelfMember();
+        if (!selfMember.hasPermission(Permission.MANAGE_ROLES)) {
+            event.reply("The bot needs the manage role permissions").setEphemeral(true).queue();
+            logger.warn("The bot needs the manage role permissions");
+            return;
+        }
+
+        SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+        boolean ephemeral = false;
+
+        if (Objects.equals(event.getSubcommandName(), CHOOSE_OPTION)) {
+            addMenuOptions(event, menu, "Select the roles to display", 1);
+            ephemeral = true;
+        } else {
+            addMenuOptions(event, menu, "Select your roles", 0);
+        }
+
+        // Handle Optional arguments
+        OptionMapping titleOption = event.getOption(TITLE_OPTION);
+        OptionMapping descriptionOption = event.getOption(DESCRIPTION_OPTION);
+
+        String title = handleOption(titleOption);
+        String description = handleOption(descriptionOption);
+
+        if (ephemeral) {
+            event.replyEmbeds(makeEmbed(title, description))
+                .addActionRow(menu.build())
+                .setEphemeral(true)
+                .queue();
+        } else {
+            event.getChannel()
+                .sendMessageEmbeds(makeEmbed(title, description))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+        }
+    }
+
+    /**
+     * Adds role options to a selection menu
+     * <p>
+     *
+     * @param event the {@link SlashCommandEvent}
+     * @param menu the menu to add options to {@link SelectionMenu.Builder}
+     * @param placeHolder the placeholder for the menu {@link String}
+     * @param minValues the minimum number of selections. nullable {@link Integer}
+     */
+    private static void addMenuOptions(@NotNull SlashCommandEvent event,
+            @NotNull SelectionMenu.Builder menu, @NotNull String placeHolder,
+            @Nullable Integer minValues) {
+
+        Role highestBotRole =
+                Objects.requireNonNull(event.getGuild()).getSelfMember().getRoles().get(1);
+        List<Role> guildRoles = Objects.requireNonNull(event.getGuild()).getRoles();
+
+        List<Role> roles = new ArrayList<>(
+                guildRoles.subList(guildRoles.indexOf(highestBotRole) + 1, guildRoles.size()));
+
+        if (minValues != null) {
+            menu.setMinValues(minValues);
+        }
+
+        menu.setPlaceholder(placeHolder).setMaxValues(roles.size());
+
+        for (Role role : roles) {
+            if (role.getName().equals("@everyone") || role.getTags().getBotId() != null)
```

Comment:
You should add brackets to this if
Although it's easy to understand, it is a bit error-prone if someone in the future tries to add conditions to it before the continue and doesn't notice the brackets

Reply:
I was surprised that this doesn't get picked up by our static analyzers. So I activated this rule: [Control structures should use curly braces](https://sonarcloud.io/organizations/togetherjava/rules?open=java%3AS121&rule_key=java%3AS121). :+1:

It's a major issue, so it doesn't fail the build, but it will be visible in bot response and sonarcloud dashboard... speaking of bot, why bot didn't respond to this PR… :(

Reply:
Same here.
Another reason for this change, just so `JJeeff248` understands, is consistency.
Since we use curly brackets everywhere, it's usually a good practice to be consistent in the way you structure your code

Reply:
All added, it makes sense to keep the code consistent, I'll keep it in mind for future :)

Fixed in b76a15fb4e674f7cef9645f756a25ff188181d53